### PR TITLE
ISPRK-15 Update to Spark 1.5

### PR DESCRIPTION
**Pull request by ghost**
https://issues.jboss.org/browse/ISPRK-15
- Updated to Spark 1.5.2
- Fixed the race condition in the streaming tests introduced by the version upgrade
- Introduced docker-compose to run the demos, which uses separate (and tagged on version) docker containers for the Infinispan Server and Spark, as opposed to use a single container for both.

### CODE REVIEW

File: examples/twitter/src/main/java/org/infinispan/spark/examples/twitter/SQLAggregationJava.java [hunk 1]
```diff
@@ -24,6 +24,8 @@
 public class SQLAggregationJava {
 
    public static void main(String[] args) throws IOException, InterruptedException, URISyntaxException {
+      String infinispanHost = args[4];
```
Comment (danberindei) [inline examples/twitter/src/main/java/org/infinispan/spark/examples/twitter/SQLAggregationJava.java]: No usage message here?

Comment (ghost) [inline examples/twitter/src/main/java/org/infinispan/spark/examples/twitter/SQLAggregationJava.java]: I will add it here and to the other samples missing.

File: src/test/scala/org/infinispan/spark/suites/StreamingSuite.scala [hunk 2]
```diff
@@ -47,9 +47,8 @@ class StreamingSuite extends FunSuite with SparkStream with MultipleServers with
       stream.writeToInfinispan(getProperties)
 
       ssc.start()
-      ssc.awaitTerminationOrTimeout(2000)
 
-      cache.size shouldBe 3
+      waitForCondition(() => cache.size == 3)
```
Comment (danberindei) [inline src/test/scala/org/infinispan/spark/suites/StreamingSuite.scala:51]: I think scalatest already has something built in for this:

http://www.artima.com/docs-scalatest-2.0.M5/org/scalatest/concurrent/Eventually.html

Comment (ghost) [inline src/test/scala/org/infinispan/spark/suites/StreamingSuite.scala:51]: This retry logic is used in other places where using Scalatest is not convenient like on a java test `org.infinispan.spark.JavaStreamApiTest`

**Comment by ghost** [comment]
@danberindei Updated

**Comment by danberindei** [comment]
Integrated, thanks Gustavo!

**Pull request closed without merging**